fcrepo-clients/src/main/java/edu/unc/lib/dl/acl/fcrepo4/ObjectPermissionEvaluator.java

```diff
@@ -0,0 +1,184 @@
+	 *            Set of principals for the agent requesting permission
+	 * @param permission
+	 *            Permission requested by agent
+	 * @return
+	 */
+	public boolean hasStaffPermission(PID pid, Set<String> agentPrincipals,
+			Permission permission) {
+		if (permission == null || pid == null || agentPrincipals == null) {
+			throw new IllegalArgumentException("Parameters must not be null");
+		}
+		if (agentPrincipals.size() == 0) {
+			return false;
+		}
+
+		Map<String, Set<String>> objectPrincipalRoles = aclFactory.getPrincipalRoles(pid);
+
+		// Check staff principals against permissions granted by roles assigned to the object
+		return agentPrincipals.stream().anyMatch(p -> {
+			return objectPrincipalRoles.containsKey(p)
+					&& objectPrincipalRoles.get(p).stream()
+					.anyMatch(r -> staffRolesToPermissions.containsKey(r)
+							&& staffRolesToPermissions.get(r).contains(permission.name()));
+		});
+	}
+
+	/**
+	 * Returns a subset of agent principals which are granted the requested
+	 * patron permission on the specified object
+	 * 
+	 * @param pid
+	 *            PID identifying the object to evaluate permissions against
+	 * @param agentPrincipals
+	 *            Set of principals for the agent requesting permission
+	 * @param permission
+	 *            Permission requested by agent
+	 * @return
+	 */
+	public Set<String> getPatronPrincipalsWithPermission(PID pid, Set<String> agentPrincipals,
+			Permission permission) {
+		if (permission == null || pid == null || agentPrincipals == null) {
+			throw new IllegalArgumentException("Parameters must not be null");
+		}
+		if (agentPrincipals.size() == 0) {
+			return Collections.emptySet();
+		}
+
+		Map<String, Set<String>> objectPrincipalRoles = aclFactory.getPrincipalRoles(pid);
+
+		// Get a list of patron principals which are granted the requested permission
+		return agentPrincipals.stream().filter(p -> {
+			return objectPrincipalRoles.containsKey(p)
+					// Check if any roles for this principal grant the requested permission
+					&& objectPrincipalRoles.get(p).stream()
+					.anyMatch(r -> patronRolesToPermissions.containsKey(r)
+							&& patronRolesToPermissions.get(r).contains(permission.name()));
+		}).collect(Collectors.toSet());
+	}
+
+	/**
+	 * Returns true if the provided principal has patron level access to the
+	 * object represented by the given pid, and false if access is revoked
+	 * 
+	 * @param pid
+	 * @param agentPrincipal
+	 * @return
+	 */
+	public boolean hasPatronAccess(PID pid, String agentPrincipal, Permission permission) {
+		return hasPatronAccess(pid, Arrays.asList(agentPrincipal), permission);
+	}
+
+	/**
+	 * Returns true if the provided agents have patron level access to the
+	 * object represented by the given pid, and false if access is revoked
+	 * 
+	 * @param pid
+	 * @param agentPrincipals
+	 * @param permission
+	 * @return
+	 */
+	public boolean hasPatronAccess(PID pid, Collection<String> agentPrincipals,
```

Comment:
We could, yes.  I just did it this way to make it more readable in the code where there was only one principal to check.  It could be easily removed if you think that would make sense